integration/test/ParseQueryTest.js

```diff
@@ -1801,4 +1805,69 @@ describe('Parse Query', () => {
       done();
     });
   });
+
+  const controllers = [
+    { name: 'In-Memory', file: require(path.resolve(__dirname, '../../lib/node/LocalDatastoreController.default')) },
+    { name: 'LocalStorage', file: require(path.resolve(__dirname, '../../lib/node/LocalDatastoreController.localStorage')) }
+  ];
+
+  for (let i = 0; i < controllers.length; i += 1) {
+    const controller = controllers[i];
+
+    describe(`Parse Query Pinning (${controller.name})`, () => {
+      beforeEach(() => {
+        const StorageController = controller.file;
+        Parse.CoreManager.setLocalDatastoreController(StorageController);
+      });
+
+      it(`${controller.name} can query from pin with name`, async () => {
+        const obj1 = new TestObject({ field: 1 });
+        const obj2 = new TestObject({ field: 2 });
+        const obj3 = new TestObject({ field: 3 });
+        const item = new Item();
+        const objects = [obj1, obj2, obj3, item];
+        await Parse.Object.saveAll(objects);
+
+        Parse.Object.pinAllWithName('test_pin', objects);
+        const query = new Parse.Query(TestObject);
+        query.greaterThan('field', 1);
+        query.fromPinWithName('test_pin');
+        const results = await query.find();
+
+        assert.equal(results.length, 2);
+        assert(results[0].get('field') > 1);
+        assert(results[1].get('field') > 1);
+      });
+
+      it(`${controller.name} can query from local datastore`, async () => {
+        const obj1 = new TestObject({ field: 1 });
+        const obj2 = new TestObject({ field: 2 });
+        const obj3 = new TestObject({ field: 3 });
+        const objects = [obj1, obj2, obj3];
+        await Parse.Object.saveAll(objects);
+
+        Parse.Object.pinAll(objects);
```

Comment:
when you call `pinAll` does it replaces the content of the default pin or does it add elements to the existing pinned elements?

Reply:
replaces